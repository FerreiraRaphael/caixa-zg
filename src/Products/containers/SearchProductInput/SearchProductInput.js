import React from 'react';
import PropTypes from 'prop-types';

import Input from '../../../Shared/components/Input';

const SearchProductInputPropTypes = {
  /**
   * Função para buscar produtos.
   */
  fetchProducts: PropTypes.func.isRequired
};

/**
 * Input de busca de produtos, faz chamada para api
 * no evento de mudança.
 */
class SearchProductInput extends React.Component {
  state = {
    value: ''
  };

  handleChange = event => {
    const value = event.target.value;
    this.setState({ value });
    this.props.fetchProducts({ query: value });
  };

  handleFocus = () => {
    this.props.focusSearchInput(true);
  };

  handleBlur = () => {
    this.props.focusSearchInput(false);
  };

  render() {
    const { loading, focus } = this.props;
    const { value } = this.state;

    return (
      <Input
        fullWidth
        label="Buscar por Nome ou SKU"
        loading={loading}
        onChange={this.handleChange}
        value={value}
        focus={focus}
        onFocus={this.handleFocus}
        onBlur={this.handleBlur}
      />
    );
  }
}

SearchProductInput.propTypes = SearchProductInputPropTypes;

export default SearchProductInput;
