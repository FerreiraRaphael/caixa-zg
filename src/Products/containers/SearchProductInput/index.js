import { connect } from "react-redux";

import SearchProductInput from "./SearchProductInput";
import { fetchProducts } from "../../modules/products";
import { focusSearchInput } from "../../../Shared/modules/app";

const mapStateToProps = state => {
  return {
    loading: state.products.fetching,
    focus: state.app.searchInputFocus
  };
};

const mapDispatchToProps = dispatch => {
  return {
    focusSearchInput: toggle => {
      dispatch(focusSearchInput(toggle));
    },
    fetchProducts: filter => {
      dispatch(fetchProducts(filter));
    }
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(SearchProductInput);
