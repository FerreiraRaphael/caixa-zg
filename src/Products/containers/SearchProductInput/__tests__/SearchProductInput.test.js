import React from "react";
import { shallow, mount } from "enzyme";

import SearchProductInput from "../SearchProductInput";

describe("SearchProductInput Component", () => {
  const setup = (propsOverride = {}, mounting = false) => {
    const props = {
      fetchProducts: jest.fn(),
      ...propsOverride
    };

    const mountFunction = mounting ? mount : shallow;

    const wrapper = mountFunction(<SearchProductInput {...props} />);

    return {
      props: wrapper.props(),
      wrapper,
      input: wrapper.find("input")
    };
  };

  it("should render", () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it("should call fetchProducts on change", () => {
    const { wrapper, input, props } = setup(null, true);
    input.simulate("change", { target: { value: "query" } });
    expect(props.fetchProducts).toBeCalledWith({ query: "query" });
  });
});
