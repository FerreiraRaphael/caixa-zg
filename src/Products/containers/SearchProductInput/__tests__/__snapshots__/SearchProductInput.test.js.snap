// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchProductInput Component should render 1`] = `
<CommonInput
  autoFocus={false}
  fullWidth={true}
  label="Buscar por Nome ou SKU"
  loading={false}
  onChange={[Function]}
  value=""
/>
`;
